File: client/newsapp-client/src/models/newsPage.ts
import { Article } from "./article";
import { map } from "rxjs/operators";

export class NewsPage {
    public country : String;
    public category : String;
    public articles : Article[];

    constructor(newsPageJSON){
        this.country = newsPageJSON.country;
        this.category = newsPageJSON.category;
        this.articles = newsPageJSON.articles.map((article : Article) => new Article(article));
    }
}

File: server/src/main/java/com/krokogator/newsapp/model/NewsPage.java
package com.krokogator.newsapp.model;

import java.util.List;

public class NewsPage {
    private String country;
    private String category;
    private List<IArticleAdapter> articles;

    public NewsPage(String country, String category, List<IArticleAdapter> articles) {
        this.country = country;
        this.category = category;
        this.articles = articles;
    }

    public String getCountry() {
        return country;
    }

    public String getCategory() {
        return category;
    }

    public List<IArticleAdapter> getArticles() {
        return articles;
    }
}
